src/Plugin/EntityBrowser/Widget/View.php

```diff
@@ -199,11 +164,11 @@ public static function processCheckbox(&$element, FormStateInterface $form_state
    * {@inheritdoc}
    */
   public function submit(array &$element, array &$form, FormStateInterface $form_state) {
-    $selected_rows = array_keys(array_filter($form_state->getValue('entity_browser_select')));
+    $selected_rows = array_values(array_filter($form_state->getUserInput()['entity_browser_select']));
```

Comment:
I'm not sure - this was necessary as $form_state->getValue() is only going to work on what's initially rendered, not any dynamically added form elements (like new entity_browser_select  checkboxes brought in by pagers, filters, sorts, infinite scroll, etc.). 

Could we manually filter `$type` and `$id` after the fact?